lib/sycamore/sycamore/transforms/extract_graph.py

```diff
@@ -300,6 +301,126 @@ def GraphEntityExtractorPrompt(entities, query):
+
+        sections = []
+
+        section: Optional[HierarchicalDocument] = None
+        element: Optional[HierarchicalDocument] = None
+        for child in doc.children:
+            if "relationships" not in child.data:
+                child.data["relationships"] = {}
+            if (
+                child.type == "Section-header"
+                and "text_representation" in child.data
+                and len(child.data["text_representation"]) > 0
+            ):
+                if section is not None:
+                    next = {
+                        "TYPE": "NEXT",
+                        "properties": {},
+                        "START_ID": section.doc_id,
+                        "START_LABEL": "SECTION",
+                        "END_ID": child.doc_id,
+                        "END_LABEL": "SECTION",
+                    }
+                    child.data["relationships"][str(uuid.uuid4())] = next
+                    element = None
+                rel = {
+                    "TYPE": "SECTION_OF",
+                    "properties": {},
+                    "START_ID": child.doc_id,
+                    "START_LABEL": "SECTION",
+                    "END_ID": doc.doc_id,
+                    "END_LABEL": "DOCUMENT",
+                }
+                child.data["relationships"][str(uuid.uuid4())] = rel
+                child.data["label"] = "SECTION"
+                section = child
+                sections.append(section)
+            else:
+                assert section is not None
+                if element is not None:
+                    next = {
+                        "TYPE": "NEXT",
+                        "properties": {},
+                        "START_ID": element.doc_id,
+                        "START_LABEL": "ELEMENT",
+                        "END_ID": child.doc_id,
+                        "END_LABEL": "ELEMENT",
+                    }
+                    child.data["relationships"][str(uuid.uuid4())] = next
+                rel = {
+                    "TYPE": "PART_OF",
+                    "properties": {},
+                    "START_ID": child.doc_id,
+                    "START_LABEL": "ELEMENT",
+                    "END_ID": section.doc_id,
+                    "END_LABEL": "SECTION",
+                }
+                child.data["relationships"][str(uuid.uuid4())] = rel
+                child.data["label"] = "ELEMENT"
+                element = child
+                section.data["children"].append(element)
+
+        doc.children = sections
+        return doc
+
+
+class ExtractSummaries(Map):
+    """
+    Extracts summaries from child documents
+    """
+
+    def __init__(self, child: Node, **resource_args):
+        super().__init__(child, f=ExtractSummaries.summarize_sections, **resource_args)
+
+    @staticmethod
+    def summarize_sections(doc: HierarchicalDocument) -> HierarchicalDocument:
+        if "EXTRACTED_NODES" in doc.data or not isinstance(doc, HierarchicalDocument):
```

Comment:
the latter condition is always false right? Since the function sig requires it

Reply:
yep, removed this.